example/diffusion_gmg/diffusion_equation.hpp

```diff
@@ -0,0 +1,320 @@
+    const std::size_t scratch_size_in_bytes = 0;
+    const std::size_t scratch_level = 1;
+
+    const parthenon::Indexer3D idxers[6]{
+        parthenon::Indexer3D(kb, jb, {ib.s, ib.s}),
+        parthenon::Indexer3D(kb, jb, {ib.e + 1, ib.e + 1}),
+        parthenon::Indexer3D(kb, {jb.s, jb.s}, ib),
+        parthenon::Indexer3D(kb, {jb.e + 1, jb.e + 1}, ib),
+        parthenon::Indexer3D({kb.s, kb.s}, jb, ib),
+        parthenon::Indexer3D({kb.e + 1, kb.e + 1}, jb, ib)};
+    constexpr int x1off[6]{-1, 1, 0, 0, 0, 0};
+    constexpr int x2off[6]{0, 0, -1, 1, 0, 0};
+    constexpr int x3off[6]{0, 0, 0, 0, -1, 1};
+    constexpr TE tes[6]{TE::F1, TE::F1, TE::F2, TE::F2, TE::F3, TE::F3};
+    constexpr int dirs[6]{X1DIR, X1DIR, X2DIR, X2DIR, X3DIR, X3DIR};
+    parthenon::par_for_outer(
+        DEFAULT_OUTER_LOOP_PATTERN, "SetFluxBoundaries", DevExecSpace(),
+        scratch_size_in_bytes, scratch_level, 0, pack.GetNBlocks() - 1,
+        KOKKOS_LAMBDA(parthenon::team_mbr_t member, const int b) {
+          const auto &coords = pack.GetCoordinates(b);
+          const int gid = pack.GetGID(b);
+          const int level = pack.GetLevel(b, 0, 0, 0);
+          const Real dxs[3]{coords.template Dxc<X1DIR>(), coords.template Dxc<X2DIR>(),
+                            coords.template Dxc<X3DIR>()};
+          for (int face = 0; face < ndim * 2; ++face) {
+            const Real dx = dxs[dirs[face] - 1];
+            const auto &idxer = idxers[face];
+            const auto dir = dirs[face];
+            const auto te = tes[face];
+            // Impose the zero Dirichlet boundary condition at the actual boundary
+            if (pack.IsPhysicalBoundary(b, x3off[face], x2off[face], x1off[face])) {
+              const int koff = x3off[face] > 0 ? -1 : 0;
+              const int joff = x2off[face] > 0 ? -1 : 0;
+              const int ioff = x1off[face] > 0 ? -1 : 0;
+              const int sign = x1off[face] + x2off[face] + x3off[face];
+              parthenon::par_for_inner(
+                  DEFAULT_INNER_LOOP_PATTERN, member, 0, idxer.size() - 1,
+                  [&](const int idx) {
+                    const auto [k, j, i] = idxer(idx);
+                    pack.flux(b, dir, var_t(), k, j, i) =
+                        sign * pack_mat(b, te, D_t(), k, j, i) *
+                        pack(b, var_t(), k + koff, j + joff, i + ioff) / (0.5 * dx);
+                  });
+            }
+            // Correct for size of neighboring zone at fine-coarse boundary when using
+            // constant prolongation
+            if (do_flux_dx &&
+                pack.GetLevel(b, x3off[face], x2off[face], x1off[face]) == level - 1) {
+              parthenon::par_for_inner(DEFAULT_INNER_LOOP_PATTERN, member, 0,
+                                       idxer.size() - 1, [&](const int idx) {
+                                         const auto [k, j, i] = idxer(idx);
+                                         pack.flux(b, dir, var_t(), k, j, i) /= 1.5;
```

Comment:
this is a silly thing, but `*= TWO_THIRDS` will be more efficient than `/= 1.5`.

Reply:
Got this one and a one_fourth elsewhere